### PR TITLE
net-p2p/resilio-sync-bin: new ebuild

### PR DESCRIPTION
**Pull request by SpiderX**
Resilio Sync is a proprietary peer-to-peer multiplatform file synchronization tool
It can sync files between devices on a local network, or between remote devices
over the Internet via a modified version of the BitTorrent protocol.

Base for ebuild from https://github.com/SpiderX/portage-overlay/tree/master/net-p2p/resilio-sync-bin

Gentoo-Bug: https://bugs.gentoo.org/486406
Package-Manager: Portage-2.3.6, Repoman-2.3.2

### CODE REVIEW

File: net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+MY_PN="${PN/-bin/}"
+BASE_URI="http://linux-packages.resilio.com/${MY_PN}/deb/pool/non-free/r/${MY_PN}/${MY_PN}_${PV}-1_-arch-.deb"
```
Comment (SoapGentoo) [inline net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild]: do both of these after the inherit line

File: net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+inherit pax-utils systemd tmpfiles unpacker user
+
+NAME="rslsync"
+QA_PREBUILT="usr/bin/${NAME}"
```
Comment (SoapGentoo) [inline net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild]: `QA_PREBUILT` just before `pkg_setup`

File: net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+DESCRIPTION="Resilient, fast and scalable file synchronization tool"
+HOMEPAGE="https://getsync.com/"
+SRC_URI="amd64? ( ${BASE_URI/-arch-/amd64} )
+	arm? ( ${BASE_URI/-arch-/armhf} )
+	x86? ( ${BASE_URI/-arch-/i386} )"
+LICENSE="all-rights-reserved"
```
Comment (SoapGentoo) [inline net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild]: empty line between `SRC_URI` and `LICENSE`

File: net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~x86"
+IUSE="pax_kernel systemd"
+RESTRICT="bindist mirror"
+S="${WORKDIR}"
```
Comment (SoapGentoo) [inline net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild]: empty line between `RESTRICT` and `S`

File: net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+pkg_setup() {
+	enewgroup ${NAME}
+	enewuser ${NAME} -1 -1 /var/lib/${MY_PN} ${NAME}
+}
+
+src_install() {
+	dobin "${S}"/usr/bin/${NAME}
```
Comment (SoapGentoo) [inline net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild]: drop `"${S}"/` (you're in it already)

File: net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+	use pax_kernel && pax-mark m "${ED%/}"/usr/bin/${NAME}
+
+	doman "${S}"/usr/share/man/man1/${MY_PN}.1.gz
```
Comment (SoapGentoo) [inline net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild]: same here

File: net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+		/var/log/${MY_PN}
+	fowners -R ${NAME}:${NAME} /etc/${MY_PN} /var/lib/${MY_PN} \
+		/var/log/${MY_PN}
+
+	newtmpfiles "${FILESDIR}"/${MY_PN}.tmpfile ${MY_PN}.conf
+
+	use systemd || newinitd "${FILESDIR}"/${MY_PN}.initd ${MY_PN}
+	use systemd || newconfd "${FILESDIR}"/${MY_PN}.confd ${MY_PN}
+	use systemd && systemd_dounit "${S}"/lib/systemd/system/${MY_PN}.service
+	use systemd && systemd_douserunit "${S}"/usr/lib/systemd/user/${MY_PN}.service
```
Comment (SoapGentoo) [inline net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild]: please always install all files, unless there's a good reason not to

File: net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+pkg_preinst() {
+	# Generate sample config
+	"${ED%/}"/usr/bin/${NAME} --dump-sample-config > \
+		"${ED%/}"/etc/${MY_PN}/config.json || die "generate config failed"
```
Comment (SoapGentoo) [inline net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild]: Use `${EROOT%/}` in `pkg_*` phases in general over `${ED%/}`

File: net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild (hunk 1 above)
Comment (SpiderX) [inline net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild]: Moved.

File: net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild (hunk 2 above)
Comment (SpiderX) [inline net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild]: Moved.

File: net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild (hunk 3 above)
Comment (SpiderX) [inline net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild]: Done.

File: net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild (hunk 4 above)
Comment (SpiderX) [inline net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild]: Done.

File: net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild (hunk 5 above)
Comment (SpiderX) [inline net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild]: Done.

File: net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild (hunk 6 above)
Comment (SpiderX) [inline net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild]: Done.

File: net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild (hunk 7 above)
Comment (SpiderX) [inline net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild]: Done, also removed systemd USE-flag as pointless in this case.

File: net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild (hunk 8 above)
Comment (SpiderX) [inline net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild]: I have to use ED%/ here because I create file in build image

Comment (SoapGentoo) [inline net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild]: oh right, this is still before installation, yes, sorry :+1:

File: net-p2p/resilio-sync-bin/files/resilio-sync.initd [hunk 9]
```diff
@@ -0,0 +1,41 @@
+#!/sbin/openrc-run
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+extra_commands="reload"
+
+NAME=${SVCNAME##*.}
+if [ -n "${NAME}" -a "${SVCNAME}" != "resilio-sync" ]; then
+    PID="/run/resilio-sync/resilio-sync.${NAME}.pid"
+    PNAME=$(echo ${RC_SVCNAME} | sed 's/\..*//g')
```
Comment (mgorny) [inline net-p2p/resilio-sync-bin/files/resilio-sync.initd]: Is the purpose of this to strip everything following `.`? If yes, then why not `${RC_SVCNAME%%.*}`? Plus, you can move it outta the `if` — it's going to give the same result in both cases.

File: net-p2p/resilio-sync-bin/files/resilio-sync.initd [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+    CONF_DEFAULT="/etc/conf.d/resilio-sync.${NAME}"
+else
+    PID="/run/resilio-sync/resilio-sync.pid"
+    PNAME=${RC_SVCNAME}
+    CONF_DEFAULT="/etc/conf.d/resilio-sync"
+fi
+CONF=${CONF:-${CONF_DEFAULT}}
```
Comment (mgorny) [inline net-p2p/resilio-sync-bin/files/resilio-sync.initd]: Did you mean `CONFIG`? That's what conf.d has. But it has `--config` explicitly specified, so I'm confused.

Also, I think it's marginally better to do:

    : "${CONF:=${CONF_DEFAULT}}"

File: net-p2p/resilio-sync-bin/files/resilio-sync.initd [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+EXEC=${EXEC:-/usr/bin/rslsync}
```
Comment (mgorny) [inline net-p2p/resilio-sync-bin/files/resilio-sync.initd]: Where is EXEC being defined?

File: net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild [hunk 12]
```diff
@@ -0,0 +1,77 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit pax-utils systemd tmpfiles unpacker user
+
+MY_PN="${PN/-bin/}"
```
Comment (mgorny) [inline net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild]: Why name it `-bin` just to make the ebuild harder? The `-bin` suffix is generally required only if there's a non`-bin` alternative (i.e. firefox vs firefox-bin).

File: net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+BASE_URI="http://linux-packages.resilio.com/${MY_PN}/deb/pool/non-free/r/${MY_PN}/${MY_PN}_${PV}-1_-arch-.deb"
+NAME="rslsync"
```
Comment (mgorny) [inline net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild]: Do not abuse variables like this. It makes the ebuild horribly hard to read.

File: net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild [hunk 14]
```diff
@@ -0,0 +1,77 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit pax-utils systemd tmpfiles unpacker user
+
+MY_PN="${PN/-bin/}"
+BASE_URI="http://linux-packages.resilio.com/${MY_PN}/deb/pool/non-free/r/${MY_PN}/${MY_PN}_${PV}-1_-arch-.deb"
```
Comment (mgorny) [inline net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild]: Prefer using uncommon characters for substitutions to make them stand out better, i.e. `@arch@`.

File: net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+
+DESCRIPTION="Resilient, fast and scalable file synchronization tool"
+HOMEPAGE="https://getsync.com/"
+SRC_URI="amd64? ( ${BASE_URI/-arch-/amd64} )
+	arm? ( ${BASE_URI/-arch-/armhf} )
+	x86? ( ${BASE_URI/-arch-/i386} )"
+
+LICENSE="all-rights-reserved"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~x86"
```
Comment (mgorny) [inline net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild]: Please don't add `~arm` keywords yourself. This package will probably require masking on specific ARMvX profiles, and our ARM team doesn't really like people fiddling around with the keywords.

File: net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+pkg_setup() {
+	enewgroup ${NAME}
+	enewuser ${NAME} -1 -1 /var/lib/${MY_PN} ${NAME}
+}
+
+src_install() {
+	dobin usr/bin/${NAME}
+	use pax_kernel && pax-mark m "${ED%/}"/usr/bin/${NAME}
+
+	doman usr/share/man/man1/${MY_PN}.1.gz
```
Comment (mgorny) [inline net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild]: This should bring out a big QA warning that `doman` is not allowed on compressed files. You need to decompress it first, e.g. using `unpack`.

File: net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+
+	dodir /var/log/${MY_PN}
```
Comment (mgorny) [inline net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild]: Don't abuse MY_PN in install either. More variables is not better code. It's just a lot of indirection that makes it unreadable.

File: net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+
+	newtmpfiles "${FILESDIR}"/${MY_PN}.tmpfile ${MY_PN}.conf
+
+	newinitd "${FILESDIR}"/${MY_PN}.initd ${MY_PN}
+	newconfd "${FILESDIR}"/${MY_PN}.confd ${MY_PN}
+	systemd_dounit lib/systemd/system/${MY_PN}.service
+	systemd_douserunit usr/lib/systemd/user/${MY_PN}.service
+}
+
+pkg_preinst() {
```
Comment (mgorny) [inline net-p2p/resilio-sync-bin/resilio-sync-bin-2.5.6.ebuild]: This really doesn't belong here. Why can't you do it in `src_install`?

**Comment by mgorny** [comment]
Also, please rebase against a fresh Gentoo master.

**Comment by SoapGentoo** [comment]
@SpiderX reopen once ready

**Pull request closed without merging**